Source/SwiftLintFramework/Rules/Idiomatic/VoidFunctionInTernaryConditionRule.swift

```diff
@@ -130,11 +130,26 @@ private class VoidFunctionInTernaryConditionVisitor: SyntaxVisitor, ViolationsSy
 
         violationPositions.append(node.questionMark.positionAfterSkippingLeadingTrivia)
     }
+
+    override func visitPost(_ node: UnresolvedTernaryExprSyntax) {
+        guard node.firstChoice.is(FunctionCallExprSyntax.self),
+              let parent = node.parent?.as(ExprListSyntax.self),
+              parent.last?.is(FunctionCallExprSyntax.self) == true,
+              !parent.containsAssignment,
+              let grandparent = parent.parent,
+              grandparent.is(SequenceExprSyntax.self),
+              let blockItem = grandparent.parent?.as(CodeBlockItemSyntax.self),
+              !blockItem.isImplicitReturn else {
```

Comment:
One option you could consider would be to use the `SwiftOperators` library (https://github.com/apple/swift-syntax/blob/main/Sources/SwiftOperators/SwiftOperators.docc/SwiftOperators.md) to fold all of the sequence expressions into expression trees before you get here. Then you'll get a well-structured AST where you can check the parent of the ternary operator without guessing at its role in the sequence expression.

That said, `SwiftOperators` is new and less tested than the parser, so consider my suggestion a little bit aspirational at this point :)

Reply:
Yes, I believe that's being considered in other PRs by @marcelofabri